README.md

```diff
@@ -0,0 +1,17 @@
+# Service Access Layer
+
+Provides tools to generate from XML description language and database bindings
```

Comment:
Tools to generate what? It would be a big help to clarify that here. What ts_sal does is somewhat difficult to explain (it took me awhile to get my head around it).

From my Python perspective ts_sal generates OMG IDL files for use with OpenSplice dds.
But of course it does much more than that, because it also generates C++, Java and LabVIEW libraries for sending and receiving OpenSplice DDS topics.

Reply:
Make sense - thanks! I reworded the first paragraph into:

# Service Access Layer package

Provides tools to turn ts_xml interface description into C++, Python, Java and
LabView interfaces. Turns XMLs with the interface description into Data
Distribution Service (DDS) messages and language bindings. Binding for the
following primitives are generated:

1. **Telemetry** - data flowing from one process to another
2. **Commands** - messages to do something in another process
3. **Events** - changes in process state, with extra arguments to detail what's
   going on


will push that shortly.

Reply:
Much better. Thanks! Perhaps "(DDS) schemas and language bindings"?

I'd be tempted to leave out 1, 2, 3 for several reasons:
* This is details that might be better in a larger document.
* The full list includes ackcmd topics.
* I am not comfortable with these descriptions. Telemetry is for data that changes continuously and is output at regular intervals. Events are for stochastic data, such as changes in discrete state (on/off, out of bounds, errors, in position...).
* I am not comfortable with "process". These messages communicate between different SAL components. (In the DDS world these messages are called "samples" but we don't tend to use that term in the SAL code.)

Reply:
I really would like to keep description of Telemetry, Events and Commands, since this is the core of services SAL provides (ignoring details). So what about this:

# Service Access Layer package

Provides tools to turn ts_xml interface description into C++, Python, Java and
LabView interfaces. Turns XMLs with the interface description into Data
Distribution Service (DDS) schemas and language bindings. Binding for the 
following primitives are generated:

* **Telemetry** - data that changes continuously and are published in regular
   intervals 
* **Commands** - messages to do something in another SAL compunent
* **Events** - stochastic data, such as changes in discrete state (on/off,..)

Reply:
I wish I could find a good detailed description in official T&S docs that you could link to, but I can't. So having it here makes sense.

Consider something along the lines of the following:

* Telemetry: messages that a SAL component writes at regular intervals, e.g. to report data that varies continuously or slowly over time. Examples include measured temperatures and encoder readings.
* Events: messages that a SAL component writes when an event occurs, such as a change in discrete state. Examples include reporting a new commanded telescope target position, and reporting that the dome shutter is open, closed or moving. Log messages are also written as events.
* Commands: messages that a SAL component reads and acts on.

Some aspects of the above text that I feel are important:
* I think it helps to have a single viewpoint: a SAL controller or a "remote". I used the controller viewpoint because I believe it is easier to understand.
* Examples are useful. (I restrained myself and did not mention the "hearbeat" event because I think it could easily be a telemetry message, and probably would be if telemetry topics weren't lower priority).
* I would love to mention CSC summary state as an event, but didn't want to try to define CSC or summary state.

Reply:
Most of that was adresed, please see #64. I don't mind pointing to password protected documents, that's not LSST problem.